src/differentiation/vecjac_products.jl

```diff
@@ -47,31 +51,121 @@ mul!(w, L, v) # = df/du * v
 L(v, p, t; VJP_input = w)    # = df/dw * v
 L(x, v, p, t; VJP_input = w) # = df/dw * v
 ```
+
+## Allowed Function Signatures
+
+For Out of Place Functions:
+
+```julia
+f(u, p, t)  # t !== nothing
+f(u, p)     # p !== nothing
+f(u)        # Otherwise
+```
+
+For In Place Functions:
+
+```julia
+f(du, u, p, t)  # t !== nothing
+f(du, u, p)     # p !== nothing
+f(du, u)        # Otherwise
```

Comment:
Does this match jacvec?

Reply:
Where is v?

Reply:
That also drops `p` and `t` after asking from them. JacVec basically only allows `f(du, u)` and `f(u)`. So this one is a strict superset

Reply:
(I always pass in the general function and then get an error that `f` can't take `p` or `t` as inputs, which is kind of weird to do if the function takes them as inputs)

Reply:
> Where is v?

Oh I see the confusion. I meant to say function signatures for the input and not the constructed operator